controllers/bootstrap/init.go

```diff
@@ -371,6 +399,81 @@ func (b *Bootstrap) DeleteCrossplaneAndProviderSubscription(namespace string) er
 	return nil
 }
 
+// decoupling Crossplane from Kafka resources and keep Kafka resources left untouched
+func (b *Bootstrap) DeleteCrossplaneProviderSubscription(namespace string) error {
+	// Fetch all the CommonService instances
+	klog.V(2).Info("Fetch all the CommonService instances")
+	csList := util.NewUnstructuredList("operator.ibm.com", "CommonService", "v3")
+	if err := b.Client.List(ctx, csList); err != nil {
+		return err
+	}
+	uninstallProvider := false
+	// make sure all the cs instance need to uninstall provider
+	for _, cs := range csList.Items {
+		if cs.GetDeletionTimestamp() != nil {
+			continue
+		}
+		if cs.Object["spec"].(map[string]interface{})["features"] != nil &&
+			cs.Object["spec"].(map[string]interface{})["features"].(map[string]interface{})["bedrockshim"] != nil &&
+			cs.Object["spec"].(map[string]interface{})["features"].(map[string]interface{})["bedrockshim"].(map[string]interface{})["crossplaneProviderRemoval"] != nil {
+			if cs.Object["spec"].(map[string]interface{})["features"].(map[string]interface{})["bedrockshim"].(map[string]interface{})["crossplaneProviderRemoval"].(bool) {
+				uninstallProvider = true
+			} else {
+				uninstallProvider = false
+				break
+			}
+		} else {
+			uninstallProvider = false
+			break
+		}
+	}
```

Comment:
I think we might need to re-consider the logic here.

If we some CommonService CR's does not say anything about Crossplane, should we just ignore it?
For example,
1. if `.spec.features.bedrockshim` is `nil`, then we may should skip this CR.
2. If `.sepc.features.bedrock.shim.enable: false`, then we should skip `crossplaneProviderRemoval` as well.
3. From my own opinion, we should only consider `uninstallProvider`/`crossplaneProviderRemoval` when `.sepc.features.bedrock.shim.enable: enable`

Reply:
sure!